daemon/algod/api/client/restClient.go

```diff
@@ -283,12 +286,77 @@ func (client RestClient) Status() (response model.NodeStatusResponse, err error)
 	return
 }
 
-// WaitForBlock returns the node status after waiting for the given round.
-func (client RestClient) WaitForBlock(round basics.Round) (response model.NodeStatusResponse, err error) {
+// WaitForBlockAfter returns the node status after trying to wait for the given
+// round+1. This REST API has the documented misfeatures of returning after 1
+// minute, regardless of whether the given block has been reached.
+func (client RestClient) WaitForBlockAfter(round basics.Round) (response model.NodeStatusResponse, err error) {
 	err = client.get(&response, fmt.Sprintf("/v2/status/wait-for-block-after/%d/", round), nil)
 	return
 }
 
+// WaitForRound returns the node status after waiting for the given round.
+func (client RestClient) WaitForRound(round uint64, waitTime time.Duration) (status model.NodeStatusResponse, err error) {
+	timeout := time.NewTimer(waitTime)
+	for {
+		status, err = client.Status()
+		if err != nil {
+			return
+		}
+
+		if status.LastRound >= round {
+			return
+		}
+		select {
+		case <-timeout.C:
+			return model.NodeStatusResponse{}, fmt.Errorf("timeout waiting for round %v with last round = %v", round, status.LastRound)
+		case <-time.After(200 * time.Millisecond):
+		}
+	}
+}
+
+const singleRoundMaxTime = globals.MaxTimePerRound * 40
+
+// WaitForRoundWithTimeout waits for a given round to be reached. As it
+// waits, it returns early with an error if the wait time for any round exceeds
+// globals.MaxTimePerRound so we can alert when we're getting "hung" waiting.
+func (client RestClient) WaitForRoundWithTimeout(roundToWaitFor uint64) error {
+	status, err := client.Status()
+	if err != nil {
+		return err
+	}
+	lastRound := status.LastRound
+
+	// If node is already at or past target round, we're done
+	if lastRound >= roundToWaitFor {
+		return nil
+	}
+
+	roundComplete := make(chan error, 2)
+
+	for nextRound := lastRound + 1; lastRound < roundToWaitFor; nextRound++ {
```

Comment:
I think the original `<` is correct, there's a `>=` check returning nil on line 330.

Reply:
Both of you should have a look at the new implementation, which is clearer.  The check in it is `<` because we only want to leave the loop when the `lastRound` returned from a `Status()` call with `==` to the `roundToWaitFor`.